core/src/main/java/de/danoeh/antennapod/core/storage/PodDBAdapter.java

```diff
@@ -873,15 +877,16 @@ public void removeDownloadLogItems(long count) {
 
     public void setQueue(List<FeedItem> queue) {
         ContentValues values = new ContentValues();
+        long[] ids = new long[queue.size()];
         db.beginTransaction();
         db.delete(TABLE_NAME_QUEUE, null, null);
         for (int i = 0; i < queue.size(); i++) {
             FeedItem item = queue.get(i);
+            ids[i] = item.getId();
```

Comment:
Not used anymore ^^ (Stupid AS, sometimes takes ages to show unused variables)
When the transaction was commited, I used the array to the corresponding items as unplayed (it they would still be new).
When testing, I found that the thread would block after commiting the transaction, I could not submit another query - So I did the marking elsewhere...